### PR TITLE
Switch to sphinxcontrib.apidoc for automatically updating API docs

### PR DESCRIPTION
**Pull request by djhoese**
As mentioned in #1540, this adds sphinxcontrib.apidoc to your sphinx documentation generation so it automatically reloads the satpy api and has up to date modules/structure.

 - [x] Closes #1540 #1538
 - [x] Fully documented

### CODE REVIEW

**Comment by djhoese** [comment]
It is done! The website and linting complete with no errors...I hope. Tests do not run on github unless linting passes. The website is executed and if there are any warnings they are raised as an error.

File: setup.cfg [hunk 1]
```diff
@@ -15,3 +15,6 @@ universal=1
 max-line-length = 120
 exclude =
     satpy/readers/li_l2.py
+    satpy/readers/scatsat1_l2b.py
```
Comment (mraspaud) [inline setup.cfg:18]: Why?

**Review by mraspaud — APPROVED** [review]
LGTM, just one inline question

**Comment by mraspaud** [comment]
Are we running doctests on satpy btw?

File: setup.cfg (hunk 1 above)
Comment (djhoese) [reply setup.cfg:18]: This is an old reader that hasn't been updated yet and has a lot of style issues (same for li_l2.py and the others).

**Comment by djhoese** [comment]
We are not and I'm not sure they are set up to be doctested (using non-existent paths, etc).

File: setup.cfg (hunk 1 above)
Comment (djhoese) [reply setup.cfg:18]: To clarify more, these modules use the deprecated Dataset import. To fully fix the styling in them I would have to update them to use xarray as flake8 (with all the various plugins we use) will likely complain about things in some way.

**Comment by sfinkens** [comment]
Awesome, thanks!

**Pull request merged**